src/Resources/contao/widgets/PickerWidget.php

```diff
@@ -0,0 +1,240 @@
+
+			return $this->multiple ? $arrValue : $arrValue[0];
+		}
+	}
+
+
+	/**
+	 * Check the selected value
+	 *
+	 * @param mixed $varInput
+	 */
+	protected function checkValue($varInput)
+	{
+		if ($varInput == '' || !is_array($this->rootNodes) || !$this->Database->fieldExists('pid', $this->foreignTable))
+		{
+			return;
+		}
+
+		if (strpos($varInput, ',') === false)
+		{
+			$arrIds = array(intval($varInput));
+		}
+		else
+		{
+			$arrIds = array_map('intval', array_filter(explode(',', $varInput)));
+		}
+
+		if (count(array_diff($arrIds, array_merge($this->rootNodes, $this->Database->getChildRecords($this->rootNodes, $this->foreignTable)))) > 0)
+		{
+			$this->addError($GLOBALS['TL_LANG']['ERR']['invalidRecords']);
+		}
+	}
+
+
+	/**
+	 * Generate the record label
+	 *
+	 * @param array $data
+	 *
+	 * @return string
+	 */
+	protected function generateRecordLabel(array $data)
+	{
+		// Label callback
+		if (is_array($this->label_callback))
+		{
+			$callback = $this->label_callback;
+
+			$label = static::importStatic($callback[0])->{$callback[1]}($data, $this->objDca);
+		}
+		elseif (is_callable($callback = $this->label_callback))
+		{
+			$label = $callback($data, $this->objDca);
+		}
+		elseif ($this->labelField)
+		{
+			// Label field
+			$label = $data[$this->labelField];
+		}
+		else
+		{
+			$label = $data['id'];
+		}
+
+		// Add the icon
+		if ($this->icon)
+		{
+			$label = sprintf('%s %s', \Image::getHtml($this->icon, '', 'style="vertical-align:middle;"'), $label);
+		}
+
+		return $label;
+	}
+
+
+	/**
+	 * Generate the widget and return it as string
+	 *
+	 * @return string
+	 */
+	public function generate()
+	{
+		$arrValues = array();
+
+		// Generate the records
+		if (!empty($this->varValue)) // Can be an array
+		{
+			$ids = array_map('intval', (array) $this->varValue);
+			$records = $this->Database->execute("SELECT * FROM {$this->foreignTable} WHERE id IN (" . implode(',', $ids) . ")" . ($this->sortable ? (" ORDER BY " . $this->Database->findInSet('id', $ids)) : ""));
+
+			while ($records->next())
+			{
+				$arrValues[$records->id] = $this->generateRecordLabel($records->row());
+			}
+		}
+
+		$return = '<input type="hidden" name="'.$this->strName.'" id="ctrl_'.$this->strId.'" value="'.implode(',', array_keys($arrValues)).'">
+  <div class="selector_container">' . (($this->sortable && count($arrValues) > 1) ? '
+    <p class="sort_hint">' . $GLOBALS['TL_LANG']['MSC']['dragItemsHint'] . '</p>' : '') . '
+    <ul id="sort_'.$this->strId.'" class="'.($this->sortable ? 'sortable' : '').'">';
+
+		foreach ($arrValues as $k=>$v)
+		{
+			$return .= '<li data-id="'.$k.'">'.$v.'</li>';
+		}
+
+		$return .= '</ul>';
+
+		if (!\System::getContainer()->get('contao.picker.builder')->supportsContext($this->context))
+		{
+			$return .= '
+	<p><button class="tl_submit" disabled>'.$GLOBALS['TL_LANG']['MSC']['changeSelection'].'</button></p>';
+		}
+		else
+		{
+			$extras = ['fieldType' => $this->fieldType];
```

Comment:
`fieldType` is an optional property, default is a radio button (see my previous comment)

Reply:
Should we fix it in the page tree widget as well then? https://github.com/contao/core-bundle/blob/master/src/Resources/contao/widgets/PageTree.php#L232

Reply:
Well I guess so. We already changed it in the picker (https://github.com/contao/core-bundle/blob/master/src/Picker/PagePickerProvider.php#L62-L67) so I guess we just missed that.

Reply:
Changed in 072b178.